### PR TITLE
Add support for empty `gpus` list to run on CPU

### LINKED ISSUE #10245: Empty list for gpus throws MisconfigurationException instead of running on CPU
## 🐛 Bug

When setting gpus to an empty list, it throws a MisconfigurationException instead of just not using a gpu, as I would expect.

### To Reproduce

Run any model with `Trainer(gpus=[])`

### Expected behavior

runs on CPU instead of crashing

### Environment
N/A

### Additional context
N/A

### PR DESCRIPTION
**Pull request by victorjoos**
## What does this PR do?
 Add a check to set gpus to None if it was initially an empty list
Fixes #10245

### Does your PR introduce any breaking changes? If yes, please list them.

None

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes? (if necessary)
- [x] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [x] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

File: pytorch_lightning/utilities/device_parser.py [hunk 1]
```diff
@@ -70,7 +70,12 @@ def parse_gpu_ids(gpus: Optional[Union[int, str, List[int]]]) -> Optional[List[i
     _check_data_type(gpus)
 
     # Handle the case when no gpus are requested
-    if gpus is None or isinstance(gpus, int) and gpus == 0 or str(gpus).strip() == "0":
+    if (
+        gpus is None
+        or (isinstance(gpus, int) and gpus == 0)
+        or str(gpus).strip() == "0"
+        or (isinstance(gpus, list) and len(gpus) == 0)
```
Comment (awaelchli) [inline pytorch_lightning/utilities/device_parser.py:77]: an alternative would be to combine
```suggestion
        or str(gpus).strip() in ("0", "[]")
```
but just a remark, this is also fine.

**Review by awaelchli — APPROVED** [review]
Thanks @victorjoos 
Could you also add a changelog entry in the CHANGELOG.md file?

**Review by ethanwharris — APPROVED** [review]
LGTM :smiley:

**Review by rohitgr7 — APPROVED** [review]

**Review by rohitgr7 — APPROVED** [review]

**Review by Borda — APPROVED** [review]

**Review by tchaton — APPROVED** [review]
LGMT !
